music_assistant/server/providers/soundcloud/soundcloudpy/asyncsoundcloudpy.py

```diff
@@ -0,0 +1,343 @@
+"""Asyncio.
+
+This file is based on soundcloudpy from Naím Rodríguez https://github.com/naim-prog
+Original package https://github.com/naim-prog/soundcloud-py
+"""
+from __future__ import annotations
+
+import aiohttp
+from aiohttp.client import ClientSession
+from attr import dataclass
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://api-v2.soundcloud.com"
+
+
+@dataclass
+class SoundcloudAsync:
+    """Soundcloud."""
+
+    o_auth: str
+    client_id: str
+    headers = None
+    app_version = None
+    firefox_version = None
+    request_timeout: float = 8.0
+
+    session: ClientSession | None = None
+
+    async def get(self, url, headers=None, params=None):
+        """Async get."""
+        async with aiohttp.ClientSession(headers=headers) as session:
```

Comment:
please do not create a new session over and over but use the shared session from the mass instance.
feel free to pass it in from above

Reply:
Can you explain a little more?

Reply:
you are creating a new aiohttp session with every get request which is inefficient.
better to use the global aiohttp session, so pass in the mass.http_session from top level to this helper class and then call: async with self.http_session 

I can adjust it later once merged if you want.